### PR TITLE
Fix pruners for DataParallel support

### PR DESCRIPTION
**Pull request by chicm-ms**

### CODE REVIEW

File: examples/model_compress/fpgm_torch_mnist.py [hunk 1]
```diff
@@ -27,8 +28,14 @@ def _get_conv_weight_sparsity(self, conv_layer):
...
 
     def print_conv_filter_sparsity(self):
-        conv1_data = self._get_conv_weight_sparsity(self.conv1)
-        conv2_data = self._get_conv_weight_sparsity(self.conv2)
+        if isinstance(self.conv1, nn.Conv2d):
+            conv1_data = self._get_conv_weight_sparsity(self.conv1)
+            conv2_data = self._get_conv_weight_sparsity(self.conv2)
+        else:
+            # self.conv1 is wrapped as PrunerModuleWrapper
+            conv1_data = self._get_conv_weight_sparsity(self.conv1.module)
```
Comment (QuanluZhang) [inline examples/model_compress/fpgm_torch_mnist.py]: a little confused, you get sparsity from weight rather than mask?

File: src/sdk/pynni/tests/test_compressor.py [hunk 2]
```diff
@@ -138,7 +138,6 @@ def test_torch_fpgm_pruner(self):
         masks = pruner.calc_mask(layer, config_list[0], if_calculated=torch.tensor(0))
         assert all(torch.sum(masks['weight'], (1, 2, 3)).numpy() == np.array([45., 45., 45., 45., 0., 0., 45., 45., 45., 45.]))
 
-        pruner.update_epoch(1)
```
Comment (QuanluZhang) [inline src/sdk/pynni/tests/test_compressor.py]: what is the reason of removing this line?

File: src/sdk/pynni/nni/compression/torch/pruners.py [hunk 3]
```diff
@@ -165,9 +170,8 @@ def update_epoch(self, epoch):
 
         if epoch > 0:
             self.now_epoch = epoch
-            for k in self.if_init_list.keys():
-                self.if_init_list[k] = True
-
+            for wrapper in self.get_modules_wrapper():
+                wrapper.registered_buffers['if_calculated'].copy_(torch.tensor(0)) # pylint: disable=not-callable
```
Comment (QuanluZhang) [inline src/sdk/pynni/nni/compression/torch/pruners.py]: which one is better? `wrapper.registered_buffers['if_calculated']` or `wrapper.if_calculated`

File: examples/model_compress/fpgm_torch_mnist.py (hunk 1 above)
Comment (chicm-ms) [reply examples/model_compress/fpgm_torch_mnist.py]: In this way, we can ensure the weight is actually pruned. It can also be calculated from mask.

File: src/sdk/pynni/nni/compression/torch/pruners.py (hunk 3 above)
Comment (chicm-ms) [reply src/sdk/pynni/nni/compression/torch/pruners.py]: Just tested, `wrapper.if_calculated` does not work,  there is no error reported, but the value is still `1` we use  `wrapper.if_calculated.copy_(torch.tensor(0))`

File: src/sdk/pynni/tests/test_compressor.py (hunk 2 above)
Comment (chicm-ms) [reply src/sdk/pynni/tests/test_compressor.py]: We circumvent wrapper to directly call cal_mask in UT.  FPGM pruner call wrapper in update epoch.

File: examples/model_compress/fpgm_torch_mnist.py (hunk 1 above)
Comment (QuanluZhang) [reply examples/model_compress/fpgm_torch_mnist.py]: got it, thanks
